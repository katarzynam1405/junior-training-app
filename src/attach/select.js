/**
 * allow editing next list in the aplication
 * @param {string} options
 * @param {Object} select
 */

export default function attachSelect(options, select) {
    document.querySelector(select.outAttr).innerHTML = options.join('');

    const input = document.querySelector(select.inAttr);

    input.removeAttribute('disabled');
    input.removeAttribute('placeholder');
}
