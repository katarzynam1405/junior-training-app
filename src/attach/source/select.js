export default function attachSourceSelect(inputList, containerAttr, searchResult) {
    document.querySelector(`[${containerAttr}]`).innerHTML = inputList.join('');
    document.querySelector(`[${searchResult}]`).innerHTML = '';
}
