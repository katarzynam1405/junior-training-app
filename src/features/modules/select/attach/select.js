export default function attachSelect(options, select) {
    document.querySelector(select.outAttr).innerHTML = options.join('');

    const input = document.querySelector(select.inAttr);
    input.removeAttribute('disabled');
    input.removeAttribute('placeholder');
}
