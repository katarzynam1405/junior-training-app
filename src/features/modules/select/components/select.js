// eslint-disable-next-line
import React from 'react';
/**
 * generates array of strings
 * @param {array} options list of strings with <li> tag
 * @param {string} template string with <li> tag
 * @return list with li elements saved as a strings
 */

export default function Select(options) {
    return (<ul>{options.map(i => <li>{i.name}</li>)}</ul>);
}

