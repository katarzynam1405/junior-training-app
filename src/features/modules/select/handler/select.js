/**
 * returns promise
 * @param {Object} params
 * @param {Object} nextElement
 * @callback callback
 */

export default function handlerSelect(params, nextElement, callback) {
    nextElement.load(params)
        .then(list => {
            nextElement.attach(
                nextElement.components(list, nextElement.template),
                nextElement
            );

            if (typeof callback === 'function') {
                callback(nextElement, list);
            }
        });
}
