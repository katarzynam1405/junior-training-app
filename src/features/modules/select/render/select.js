/**
 * generates array of strings
 * @param {array} options list of strings with <li> tag
 * @param {string} template string with <li> tag
 * @return list with li elements saved as a strings
 */

export default function renderSelect(options, template) {
    const result = [];

    const replacements = { key: 'name' };
    const getValue = (tag, option) => {
        let val = '';
        for (const key in replacements) {
            if (key === tag) {
                val = option[replacements[key]];
                break;
            }
        }
        return val;
    };

    options.forEach(option => result.push(template.replace(/{(\w+)}/g,
        (match, tag) => getValue(tag, option))));

    return result;
}
