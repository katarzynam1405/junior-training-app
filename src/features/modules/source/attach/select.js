/**
 * resets table content
 * @param {Object} inputList
 * @param {string} containerAttr
 * @param {string} searchResult
 */
export default function attachSourceSelect(inputList, containerAttr, searchResult) {
    document.querySelector(containerAttr).innerHTML = inputList.join('');
    document.querySelector(searchResult).innerHTML = '';
}

