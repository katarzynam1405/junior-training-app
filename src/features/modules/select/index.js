import attach from './attach';
import handler from './handler';
import render from './render';

export default {
    attach, handler, render
};
