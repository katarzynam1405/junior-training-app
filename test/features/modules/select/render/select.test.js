import select from '../../../../../src/features/modules/select/render/select';

describe('f/m/select/render/select', () => {
    it('should return array of strings', () => {
        const results = select([
            { name: '1234' },
            { name: 'xyzw' }
        ], 'abc-{key}');

        expect(results).toMatchSnapshot();
    });
});
