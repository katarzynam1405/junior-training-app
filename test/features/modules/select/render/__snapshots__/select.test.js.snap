// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render/select should return array of strings 1`] = `
Array [
  "abc-1234",
  "abc-xyzw",
]
`;
